### PR TITLE
OCM-10813 | ci: Update the steps on prow for rosacli CI with shared functions

### PR DESCRIPTION
**Pull request by xueli181114**

### CODE REVIEW

File: ci-operator/step-registry/rosa/tests/pull-request/rosa-tests-pull-request-commands.sh [hunk 1]
```diff
@@ -6,52 +6,87 @@ set -o pipefail
...
+declare -a run_times=(
+  "day1-post"
+  "day2"
+  "destructive"
+  "destroy" 
+  "destroy-post"
+)
+for run_time in "${run_times[@]}"; do
+  run_testing_steps $run_time
+done
```
Comment (yasun1) [inline ci-operator/step-registry/rosa/tests/pull-request/rosa-tests-pull-request-commands.sh:92]: Miss a newline.

Comment (xueli181114) [reply ci-operator/step-registry/rosa/tests/pull-request/rosa-tests-pull-request-commands.sh]: Updated

**Comment by xueli181114** [comment]
/pj-rehearse pull-ci-openshift-rosa-master-e2e-presubmits-pr-rosa-hcp-advanced

**Comment by xueli181114** [comment]
/pj-rehearse periodic-ci-openshift-rosa-master-e2e-rosa-hcp-advanced-critical-high-f3

**Comment by xueli181114** [comment]
/pj-rehearse pull-ci-openshift-rosa-master-e2e-presubmits-pr-rosa-hcp-advanced

**Comment by xueli181114** [comment]
/pj-rehearse periodic-ci-openshift-rosa-master-e2e-rosa-hcp-advanced-critical-high-f3

**Comment by xueli181114** [comment]
/pj-rehearse pull-ci-openshift-rosa-master-e2e-presubmits-pr-rosa-hcp-advanced

**Comment by xueli181114** [comment]
/pj-rehearse pull-ci-openshift-rosa-master-e2e-presubmits-pr-rosa-hcp-advanced

**Comment by xueli181114** [comment]
/pj-rehearse pull-ci-openshift-rosa-master-e2e-presubmits-pr-rosa-hcp-advanced

**Comment by xueli181114** [comment]
/pj-rehearse periodic-ci-openshift-rosa-master-e2e-rosa-hcp-advanced-critical-high-f3

**Comment by xueli181114** [comment]
/pj-rehearse periodic-ci-openshift-rosa-master-e2e-rosa-hcp-advanced-critical-high-f3

**Comment by xueli181114** [comment]
/pj-rehearse pull-ci-openshift-rosa-master-e2e-presubmits-pr-rosa-hcp-advanced

**Comment by xueli181114** [comment]
/pj-rehearse pull-ci-openshift-rosa-master-e2e-presubmits-pr-rosa-hcp-advanced

**Comment by xueli181114** [comment]
/pj-rehearse periodic-ci-openshift-rosa-master-e2e-rosa-hcp-advanced-critical-high-f3

**Comment by xueli181114** [comment]
/pj-rehearse periodic-ci-openshift-rosa-master-e2e-rosa-hcp-advanced-critical-high-f3

**Comment by xueli181114** [comment]
/pj-rehearse pull-ci-openshift-rosa-master-e2e-presubmits-pr-rosa-hcp-advanced

**Comment by xueli181114** [comment]
/pj-rehearse periodic-ci-openshift-rosa-master-e2e-rosa-hcp-advanced-critical-high-f3

**Comment by xueli181114** [comment]
/pj-rehearse pull-ci-openshift-rosa-master-e2e-presubmits-pr-rosa-hcp-advanced

**Comment by xueli181114** [comment]
/pj-rehearse periodic-ci-openshift-rosa-master-e2e-rosa-hcp-advanced-critical-high-f3

**Comment by xueli181114** [comment]
/pj-rehearse pull-ci-openshift-rosa-master-e2e-presubmits-pr-rosa-hcp-advanced

**Comment by xueli181114** [comment]
/pj-rehearse pull-ci-openshift-rosa-master-e2e-presubmits-pr-rosa-hcp-advanced

**Comment by xueli181114** [comment]
/pj-rehearse periodic-ci-openshift-rosa-master-e2e-rosa-hcp-advanced-critical-high-f3

**Comment by xueli181114** [comment]
Previous periodic [job log](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/56178/rehearse-56178-periodic-ci-openshift-rosa-master-e2e-rosa-hcp-advanced-critical-high-f3/1831935994031509504)

**Comment by xueli181114** [comment]
/pj-rehearse pull-ci-openshift-rosa-master-e2e-presubmits-pr-rosa-hcp-advanced

**Comment by xueli181114** [comment]
/pj-rehearse periodic-ci-openshift-rosa-master-e2e-rosa-hcp-advanced-critical-high-f3

**Comment by xueli181114** [comment]
/pj-rehearse periodic-ci-openshift-rosa-master-e2e-rosa-hcp-advanced-critical-high-f3

**Comment by xueli181114** [comment]
[e2e-presubmits-pr-rosa-hcp-advanced](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/56178/rehearse-56178-pull-ci-openshift-rosa-master-e2e-presubmits-pr-rosa-hcp-advanced/1831969762100383744)
[periodic-ci-openshift-rosa-master-e2e-rosa-day1-negative-f7 ](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/56178/rehearse-56178-periodic-ci-openshift-rosa-master-e2e-rosa-day1-negative-f7/1831969846108098560)
[periodic-ci-openshift-rosa-master-e2e-rosa-hcp-advanced-critical-high-f3](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/56178/rehearse-56178-periodic-ci-openshift-rosa-master-e2e-rosa-hcp-advanced-critical-high-f3/1832027975021760512)

**Pull request merged**